mne/viz/_3d.py

```diff
@@ -353,12 +353,55 @@ def plot_trans(info, trans_fname='auto', subject=None, subjects_dir=None,
     return fig
 
 
+def _percent_to_control_points(clim, stc_data, colormap):
+    """Private helper function to convert percentiles to control points.
+
+    Note: If using 'mne_analyze', generate cmap control points for a directly
+    mirrored cmap for simplicity (i.e., no normalization is computed to account
+    for a 2-tailed mne_analyze cmap).
+
+    Parameters
+    ----------
+    clim : str | dict
+        Desired percentages used to set cmap control points.
+
+    Returns
+    -------
+    ctrl_pts : list (length 3)
+        Array of floats corresponding to values to use as cmap control points.
+    """
+
+    # Based on type of limits specified, get cmap control points
+    if clim == 'auto':
+        # Set upper and lower bound based on percent, and get average between
+        ctrl_pts = np.percentile(np.abs(stc_data), [96, 99.95])
+        ctrl_pts.insert(1, np.average(ctrl_pts))
+    elif isinstance(clim, dict):
+        # Get appropriate key for clim if it's a dict
+        limit_key = ['lims', 'pos_lims'][colormap == 'mne_analyze']
+        if limit_key not in clim.keys():
+            raise KeyError('lims" OR "pos_lims" must be defined depending on'
+                           ' if colormap is "mne_analyze"')
+        if clim['kind'] == 'percent':
+            ctrl_pts = np.percentile(np.abs(stc_data),
+                                     list(np.abs(clim[limit_key])))
+        elif clim['kind'] == 'value':
+            ctrl_pts = list(clim[limit_key])
+        else:
+            raise ValueError('If clim is a dict, clim[kind] must be '
+                             ' "value" or "percent"')
+    else:
+        raise ValueError('"clim" must be "auto", tuple, or dict')
+
+    return ctrl_pts
+
+
 def plot_source_estimates(stc, subject=None, surface='inflated', hemi='lh',
                           colormap='hot', time_label='time=%0.2f ms',
-                          smoothing_steps=10, fmin=5., fmid=10., fmax=15.,
-                          transparent=True, alpha=1.0, time_viewer=False,
+                          smoothing_steps=10, fmin=None, fmid=None, fmax=None,
+                          transparent=None, alpha=1.0, time_viewer=False,
                           config_opts={}, subjects_dir=None, figure=None,
-                          views='lat', colorbar=True):
+                          views='lat', colorbar=True, clim=None):
```

Comment:
okay, but I don't follow why `clim` must be `None` to deprecate. Why should we default to the old behavior? We should just throw a warning if people go for the old behavior, no?